spec/defines/client/config/user_spec.rb

```diff
@@ -0,0 +1,167 @@
+      end
+
+      let :params do {
+        :target  => target
+      } end
+
+      it do
+        should contain_file(target).with({
+          :ensure  => 'file',
+          :owner   => title,
+          :mode    => '0600'
+        })
+      end
+
+    end # describe 'with a user provided target'
+
+    describe 'user_home_dir behavior' do
+
+      context 'with a user provided user_home_dir' do
+
+        let :user_home_dir do
+          '/path/to/home'
+        end
+
+        context 'with manage_user_ssh_dir default value' do
+
+          let :params do {
+            :user_home_dir => user_home_dir
+          } end
+
+          it 'should contain ssh directory and ssh config' do
+
+            should contain_file("#{user_home_dir}/.ssh").with({
+              :ensure  => 'directory',
+              :owner   => title,
+              :mode    => '0700'
+            }).that_comes_before("File[#{user_home_dir}/.ssh/config]")
+
+            should contain_file("#{user_home_dir}/.ssh/config").with({
+              :ensure  => 'file',
+              :owner   => title,
+              :mode    => '0600'
+            })
+          end
+
+        end # context 'with manage_user_ssh_dir default value'
+
+        context 'with manage_user_ssh_dir set to false' do
+
+          let :params do {
+            :user_home_dir       => user_home_dir,
+            :manage_user_ssh_dir => false
+          } end
+
+          it do
+            should_not contain_file("#{user_home_dir}/.ssh")
+          end
+
+        end # context 'with manage_user_ssh_dir set to false'
+
+      end # context 'with a user provided user_home_dir'
+
+      context 'with no user provided user_home_dir' do
+        it 'with manage_user_ssh_dir default value' do
+          should contain_file("/home/#{title}/.ssh").that_comes_before("File[/home/#{title}/.ssh/config]")
+          should contain_file("/home/#{title}/.ssh/config")
+        end
+
+        context 'with manage_user_ssh_dir set to false' do
+          let :params do {
+            :manage_user_ssh_dir => false
+          } end
+
+          it do
+            should_not contain_file("/home/#{title}/.ssh")
+          end
+
+          it do
+            should contain_file("/home/#{title}/.ssh/config")
+          end
+
+        end # context 'with manage_user_ssh_dir set to false'
+
+      end # context 'with no user provided user_home_dir'
+
+    end # describe 'user_home_dir behavior'
+
+    describe 'ssh configuration content' do
+
+      let :params do {
+        :options => ssh_options
+      } end
+
+      it 'should have single value' do
+        should contain_file("/home/#{title}/.ssh/config").with({
+          :content => /HashKnownHosts\s+yes/
+        })
+      end
+
+      it 'should have Hash value' do
+        should contain_file("/home/#{title}/.ssh/config").with({
+          :content => /Host \*\.in2p3\.fr\s*\n\s+(User\s+riton|GSSAPIAuthentication\s+no)\s*\n\s+(User\s+riton|GSSAPIAuthentication\s+no)/
```

Comment:
My bad, I've just realized that you were using the `sort()` method on the options. The current regexp is unnecessarily complicated here...